src/components/signup-view/signup-view.jsx

```diff
@@ -1,55 +1,125 @@
 import { React, useState } from "react";
+import FloatingLabel from 'react-bootstrap/FloatingLabel';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
 
-export const SignupView = () => {
+export const SignupView = ({onLoggedIn}) => {
 
-    const handleSubmit = (event) => {};
     const [username, setUsername] = useState("");
     const [password, setPassword] = useState("");
     const [email, setEmail] = useState("");
-    const [birthday, setBirthday] = useState("");
+    const [birthdate, setBirthdate] = useState("");
+    const Backend_API = "https://guarded-peak-19726.herokuapp.com";
+    
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        
+            const data = {
+                username: username,
+                password: password,
+                email: email,
+                birthdate: birthdate
+              };
+              
+              fetch(`${Backend_API}/users`, {
+                method: "POST",
+                body: JSON.stringify(data),
+                headers: {
+                  "Content-Type": "application/json"
+                }
+              });
+        
+        // Perform login after successful registration
+        
+                fetch(`${Backend_API}/login`, {
+                    method: "POST",
+                    body: JSON.stringify(data),
+                    headers: {
+                      "Content-Type": "application/json"
+                    }
+                  })
+                    .then((response) => response.json())
+                    .then((data) => {
+                      console.log('Login response: ', data);
+                      if (data.user) {
+                        localStorage.setItem('user', JSON.stringify(data.user));
+                        localStorage.setItem('token', data.token);
+                        onLoggedIn(data.user, data.token);
+                        alert('Logged in successfully, please proceed to the homepage')
+                      } else {
+                        alert('No such user');
+                      }
+                    })
+                    .catch((error) => {
+                      console.log("Error occurred during login: ", error);
+                      alert("Something went wrong during login");
+                    });
+            };
```

Comment:
Hi @chrisdvadashanta ,
This is not good practice, because you making this call without knowing if the first API call ie registration was successful or not. I would be better to only login if the user has registered successfully.